### DISCUSSION TITLE
Definitations of f_{p} and g_{p} functions

### DISCUSSION THREAD

**Discussion by yingjiewei** [scikit-hep/pyhf#1959] [Q&A] [answered]
Thanks so much for all of your efforts on the pyhf. It's really amazing package and tool!
 
I have a question about the definitions of function f_{p} and g_{p} in https://pyhf.readthedocs.io/en/v0.6.3/intro.html#id25.

(1) Are they defined as the way in Sect 4.1 of https://cds.cern.ch/record/1456844/files/CERN-OPEN-2012-016.pdf?
    
(2) if they are, in pyhf, which Piecewise method is used in pyhf, I mean Piecewise Linear or Piecewise Exponential?

Thanks sooo much!

Yingjie

**Comment by kratsg** [comment, marked as answer]
1. Yes, that's correct. Let us know how the documentation should be changed to clarify this further. As you see from our [documentation](https://scikit-hep.org/pyhf/api.html#interpolators), we have the various interpolators implemented as documented in the paper.
2. The defaults are set in the [code](https://github.com/scikit-hep/pyhf/blob/f01dcf59da10ec9313345305efc49cfb7186bf99/src/pyhf/pdf.py#L209-L212) as `InterpCode=4` "Polynomial Interpolation and Exponential Extrapolation" for `normsys` (and the piecewise variant suffixed by `p` for `histosys`). This matches the default in `ROOT` as well. These defaults were changed in #1114 for `v0.6.0+`.

**Comment by yingjiewei** [comment]
Thanks so much for this help! 

To further improve the document, it would be better to add a citation in line: "makes use of interpolating functions, f_p and f_g, constructed from a small number of evaluations of the expected" of https://scikit-hep.org/pyhf/intro.html

and also add the reference on the table: https://scikit-hep.org/pyhf/intro.html#id26

The reference might be Sect 4.1 of https://cds.cern.ch/record/1456844/files/CERN-OPEN-2012-016.pdf or your code API of course!

**Reply by matthewfeickert** [reply]
We probably won't cite sections of [CERN-OPEN-2012-016](https://cds.cern.ch/record/1456844/) directly as we intentionally revised the nomenclature and notation used given the confusion people had occasionally with the original documentation. The introductory sections of the docs are mostly taken from the ATLAS PUB note [Reproducing searches for new physics with the ATLAS experiment through publication of full statistical likelihoods](https://inspirehep.net/literature/1795223), where we made the same decisions to avoid having to be tied to the original notation.

**Reply by yingjiewei** [reply]
Ah, it's definitely ok! Totally understandable!

**Reply by matthewfeickert** [reply]
Your feedback is still quite helpful though, so thank you for both your question and your time to give us your thoughts!

**Reply by yingjiewei** [reply]
Thanks for helping me as well and also the very very excellent project!
